archspec/cpu/detect.py

```diff
@@ -88,6 +90,7 @@ def proc_cpuinfo() -> Microarchitecture:
 
     if architecture == AARCH64:
         return partial_uarch(
+            name=_arm_cpu_name_by_id(data),
```

Comment:
If we want to detect by `cpuid` on ARM we could just add that single field to our JSON file. I don't think we want to add an entire new submodule and a new search mechanism for that. We should just improve the current mechanism where it can't distinguish among two microarchitectures.

Reply:
We definitely need to detect by _cpuid_ as two cores with a different cost-model (for instance, number of vector pipes) can have the same features.  Hence the definitive "what CPU is this" is actually the CPU ID.

As for how to do it..  My thought is that using an external module from Arm is indeed an extra submodule - but on the other hand, it's a definitive list from the original source.  That makes it easier to maintain via a single "pull" every few months instead of transcribing new cores into the JSON manually when they arrive, and has lower chance of error.

Reply:
What can we potentially use from upstream? Do you provide:
1. A list of features that are ensured to be on the microarchitecture
2. A list of supported compilers with flags

?

Reply:
Neither a list of features, nor a list of compiler flags are present in that repository - I'm not aware of any other source for the former other than PDF / HTML docs, and for the latter (compiler versions and flags) again is a docs (or rather, a GNU/LLVM release notes) trawling process no JSON feed.  So, yes, I think both CPU ID and the JSON file are required.

Reply:
That's certainly do-able, likely not much effort. We'll take a look. CPU IDs are a one or two per year type of edit so it's not major. 

Do you know the current behaviour for x86 - there is a dependency on some forked repo already ( https://github.com/archspec/cpuid ) - and yet I can't see those x86 CPU IDs in the JSON files.  Is that a work in progress - has a mod to the schema for microarchitectures already been proposed to bring CPU ID in?